pkg/ctl/cmdutils/configfile.go

```diff
@@ -761,3 +762,84 @@ func NewDeleteIAMServiceAccountLoader(cmd *Cmd, sa *api.ClusterIAMServiceAccount
 
 	return l
 }
+
+// NewUpdateNodegroupLoader will load config or use flags for 'eksctl update nodegroup'.
+func NewUpdateNodegroupLoader(cmd *Cmd) ClusterConfigLoader {
+	l := newCommonClusterConfigLoader(cmd)
+
+	l.validateWithConfigFile = func() error {
+		var clusterConfig *api.ClusterConfig
+		var err error
+		if clusterConfig, err = eks.LoadConfigFromFile(l.ClusterConfigFile); err != nil {
+			return err
+		}
+
+		l := len(clusterConfig.ManagedNodeGroups)
+		if l > 1 {
+			return errors.New("please update one NodeGroup at a time")
```

Comment:
so just to clarify on the UX...

say I had a config file like
```yaml
managedNodeGroups:
- name: one
  someOtherField: foobar
- name: two
  updateConfig:
    shrugEmoji: true
- name: three
```
in order to run this command i would have to edit my config file to remove those nodegroups?

Reply:
right now yes but I'm not dead set on this UX, I meant to ask about this and I'm happy to change it. do we want `update nodegroup` to be able to update multiple nodegroups at once?

Reply:
I think so yes. People are likely to have their cluster config stored somewhere so it feels weird that we would require them to remove bits to run this command. With the config above I would have to comment-run-comment 3 times which does not seem fun.

Reply:
good shout! opened a new issue to add this afterwards: https://github.com/weaveworks/eksctl/issues/3857